internal/controllers/reconciliation/overrides_test.go

```diff
@@ -349,3 +349,169 @@ func TestOverrideManagedFields(t *testing.T) {
 		return cm.Data["foo"] == "bar"
 	})
 }
+
+func TestOverrideHyphenatedFieldNames(t *testing.T) {
```

Comment:
Let's either merge all of the new tests into a single integration test, or remove them all and make sure we have enough test coverage in the parser itself.

The reason being that these tests are pretty slow and we already have pretty good coverage of the override codepath